packages/hub/src/utils/symlink.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * Heavily inspired by https://github.com/huggingface/huggingface_hub/blob/fcfd14361bd03f23f82efced1aa65a7cbfa4b922/src/huggingface_hub/file_download.py#L517
+ */
+
+import * as fs from "node:fs/promises";
+import * as path from "node:path";
+import { getHFHubCachePath } from "../lib";
+import * as os from "node:os";
+
+const MAX_CACHE_SIZE = 1000;
+const _are_symlinks_supported_in_dir = new Map<string, boolean>();
+
+export function reset(): void {
+	_are_symlinks_supported_in_dir.clear();
+}
+
+function expandUser(path: string): string {
+	if (path.startsWith("~")) {
+		return path.replace("~", os.homedir());
+	}
+	return path;
+}
+
+function commonPath(...paths: [string, ...string[]]): string {
+	// Normalize paths
+	paths = paths.map((p) => path.resolve(p)) as [string, ...string[]];
+
+	const parts = paths.map((p) => p.split(path.sep));
+	const common = parts[0];
+
+	for (let i = 1; i < parts.length; i++) {
+		for (let j = 0; j < common.length; j++) {
+			if (parts[i][j] !== common[j]) {
+				common.length = j;
+				break;
+			}
+		}
+	}
+
+	return common.join(path.sep) || path.sep;
+}
+
+/**
+ * Return whether the symlinks are supported on the machine.
+ *
+ * Since symlinks support can change depending on the mounted disk, we need to check
+ * on the precise cache folder. By default, the default HF cache directory is checked.
+ */
+async function areSymlinksSupported(cache_dir: string | undefined): Promise<boolean> {
+	if (!cache_dir) {
+		cache_dir = getHFHubCachePath();
+	}
+	cache_dir = path.resolve(expandUser(cache_dir));
+	if (!_are_symlinks_supported_in_dir.has(cache_dir)) {
+		_are_symlinks_supported_in_dir.set(cache_dir, true);
+
+		await fs.mkdir(cache_dir, { recursive: true });
+		const tmp_dir = await fs.mkdtemp(cache_dir);
+		try {
+			const src_path = path.join(tmp_dir, "dummy_src");
+			await fs.open(src_path, "w").then((fd) => fd.close());
+			const dst_path = path.join(tmp_dir, "dummy_dst");
+
+			const relative_src = path.relative(src_path, path.dirname(dst_path));
```

Comment:
```suggestion
			const relative_src = path.relative(path.dirname(dst_path), src_path);
```

I think python & JS arguments are not in the same order